library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -0,0 +1,33 @@
+package com.novoda.downloadmanager;
+
+import android.content.ContentResolver;
+
+import com.novoda.downloadmanager.lib.DownloadManager;
+
+public class DownloadManagerBuilder {
+
+    private ContentResolver contentResolver;
+    private boolean verboseLogging;
+
+    public static DownloadManagerBuilder create() {
+        return new DownloadManagerBuilder();
+    }
+
+    public DownloadManagerBuilder withVerboseLogging() {
+        this.verboseLogging = true;
+        return this;
+    }
+
+    public DownloadManagerBuilder with(ContentResolver contentResolver) {
```

Comment:
I could do it through constructor, I'd argue it's easier for clients to use the `create()` method. (I'll make the constructor package now). Through the build method also makes sense because it enforces the use of a content resolver, but just for fluid api I believe for a client to do `.create()` then `.` and see the auto fill say `with...` , `with...`, `with...` `.build()`.    

I think through build is preferred because it enforces compile time correctness (in the standard case), but with a builder client facing, I think the ease of seeing the whole api (backed by runtime checks) wins for me.

@ouchadam vote
@amlcurran vote

and we'll go with the consensus